### PR TITLE
Compatibility pass on flutter/scheduler tests for JavaScript compilation. (2)

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Flags certain test cases in scheduler using package:test tags in order to facilitate landing support for running all framework tests in a browser.

Requires: https://github.com/flutter/flutter/pull/33349

### CODE REVIEW

File: packages/flutter/test/scheduler/animation_test.dart [hunk 1]
```diff
@@ -13,7 +13,11 @@ import 'scheduler_tester.dart';
 class TestSchedulerBinding extends BindingBase with ServicesBinding, SchedulerBinding { }
 
 void main() {
-  final SchedulerBinding scheduler = TestSchedulerBinding();
+  SchedulerBinding scheduler;
+
+  setUpAll(() {
+    scheduler = TestSchedulerBinding();
+  });
```
Comment (Hixie) [inline packages/flutter/test/scheduler/animation_test.dart]: this should be semantically identical. that it isn't is concerning. can you elaborate?

File: packages/flutter/test/scheduler/benchmarks_test.dart [hunk 2]
```diff
@@ -64,5 +68,5 @@ Future<void> main() async {
       expect(endFramesBegun, equals(startFramesBegun + 1));
       expect(endFramesDrawn, equals(startFramesDrawn + 1));
     });
-  });
+  }, tags: 'web_unimplemented');
```
Comment (Hixie) [inline packages/flutter/test/scheduler/benchmarks_test.dart]: i'd prefer if we kept the number of `test` features we are using here to a minimum.

File: packages/flutter/test/scheduler/animation_test.dart (hunk 1 above)
Comment (yjbanov) [reply packages/flutter/test/scheduler/animation_test.dart]: Yeah, this looks like a bug in the web engine.

File: packages/flutter/test/scheduler/benchmarks_test.dart (hunk 2 above)
Comment (yjbanov) [reply packages/flutter/test/scheduler/benchmarks_test.dart]: Is there a good alternative in the set of features we are already using? `tags` seems like the most idiomatic way to do this (they were added for this exact use-case; I added this feature :sweat_smile: to `package:test`).

Comment (Hixie) [reply packages/flutter/test/scheduler/benchmarks_test.dart]: `skip`?

Comment (yjbanov) [reply packages/flutter/test/scheduler/benchmarks_test.dart]: My understanding is that `skip` always skips. You can't control it from the command-line. So, for example, if you fix zero or more tests that are marked with `web_unimplemented`, you still have the option to run them by passing the appropriate command-line args.

Comment (jonahwilliams) [reply packages/flutter/test/scheduler/benchmarks_test.dart]: I though about that to, but in this case if you are using `skip: isBrowser` you can just change that value to false and re-run

File: packages/flutter/test/scheduler/animation_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/test/scheduler/animation_test.dart]: I moved the test initialization logic into the generated test entrypoints and this issue went away 🎊

**Comment by jonahwilliams** [comment]
Updated with feedback and to run on CI

File: packages/flutter/test/scheduler/benchmarks_test.dart (hunk 2 above)
Comment (Hixie) [reply packages/flutter/test/scheduler/benchmarks_test.dart]: or comment out the line. why would you need to control it from the command line?

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**